Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -960,17 +960,27 @@ TIFFImageIO::ReadRawByteFromTag(unsigned int t, unsigned int & value_count)
 void
 TIFFImageIO::PopulateColorPalette()
 {
-  m_ColorPalette.resize(256);
-  for (unsigned int cc = 0; cc < 256; ++cc)
+  if (m_TotalColors > 0)
   {
-    unsigned short red, green, blue;
-    this->GetColor(cc, &red, &green, &blue);
+    // if a palette have been found store it in m_ColorPalette
+    unsigned int nColors = static_cast<unsigned int>(m_TotalColors);
+    m_ColorPalette.resize(nColors);
+    for (unsigned int cc = 0; cc < static_cast<unsigned int>(nColors); ++cc)
```

Comment:
Since `nColors` is already unsigned, this `static_cast` is not needed.

Reply:
oups!

Reply:
Your latest push c4d417f6f69e9b9f2b7209adf0c6ff0d3c7e8d49 didn't seem to have changed anything.

Reply:
reoups...